yetanotherco/aligned_layer
# feat: add node_exporter metrics

# Add node_exporter metrics

## Description
Adds [node_exporter](https://github.com/prometheus/node_exporter) to be deployed in the Aligned servers.
- Creates a script to install node_exporter.
- Creates an Ansible playbook to install node_exporter in the servers.
- Uses the [node_exporter_full](https://grafana.com/grafana/dashboards/1860-node-exporter-full) Grafana dashboard.

## How to Test
1. Install node_exporter:
```bash
make install_node_exporter
```
2. Run node_exporter:
```bash
node_exporter
```
3. Go to `localhost:9100` and you should see the exposed metrics.
4. Run metrics:
```bash
make run_metrics
```
5. Go to `localhost:3000` and you should see the new dashboard `Node Exporter Full` with your metrics displayed.
> [!NOTE]
> If you are on macos, some of the metrics will appear with `no data`.

6. Ask for a server and test the playbook.

## Type of change


## Checklist

  - [Link to the open issue addressing it]()
